### PR TITLE
add explination of header pragma to ffi section of manual

### PR DESCRIPTION
**Pull request by barcharcraz**

### CODE REVIEW

File: doc/manual/ffi.txt [hunk 1]
```diff
@@ -36,6 +36,19 @@ name is the Nim identifier *exactly as spelled*:
 Note that this pragma is somewhat of a misnomer: Other backends will provide
 the same feature under the same name.
 
+Header pragma
+-------------
+Specifies that a proc or variable name should not be declared and insted
```
Comment (reactormonk) [inline doc/manual/ffi.txt]: typo.

**Comment by Araq** [comment]
Instead port the pragmas documented in nimc.html to the manual.

**Comment by kirbyfan64** [comment]
I don't know Nim's standpoint on the commas, but shouldn't the last two sentences be written like this:

> Note that the compiler has no knowledge of the contents of this header file **,** and it is possible to have name collisions. Because of this **,** the header pragma is somewhat unsafe.

Note the bolded commas.

**Comment by barcharcraz** [comment]
I'll make requested changes tonight. I'll also update the Header pragma section in the compiler user guide with the warning on collisions.

File: doc/manual/ffi.txt [hunk 2]
```diff
@@ -36,6 +36,19 @@ name is the Nim identifier *exactly as spelled*:
...
+the generated code should contain an ``#include``:
+
+.. code-block:: NIm
```
Comment (dom96) [inline doc/manual/ffi.txt]: Lowercase ``NIm`` here please.

**Comment by Araq** [comment]
How about updating this PR to at least fix the typos?

**Pull request closed without merging**